src/server/service/kv.rs

```diff
@@ -899,21 +875,25 @@ impl<T: RaftStoreRouter + 'static> tikvpb_grpc::Tikv for Service<T> {
             return self.send_fail_status_to_stream(ctx, sink, error, code);
         }
 
-        let stream = stream
-            .map(|resp| (resp, WriteFlags::default().buffer_hint(true)))
-            .map_err(|e| {
-                let code = RpcStatusCode::Unknown;
-                let msg = Some(format!("{:?}", e));
-                GrpcError::RpcFailure(RpcStatus::new(code, msg))
-            });
+        let future = stream_future
```

Comment:
This requires addition poll to actually get the first response. How about make `cb` returns a Sink instead of accept a `Stream`?

Reply:
Yes, I also made this mistake. It will causes more  context-switch, which is a big performance lost for little query. So please make sure the grpc threads can get the stream content easily.

Reply:
ok